crystal-map-connector/src/main/kotlin/ch/ergon/dope/extension/type/relational/SatisfiesSchemaExpression.kt

```diff
@@ -0,0 +1,127 @@
+package ch.ergon.dope.extension.type.relational
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.asArrayField
+import ch.ergon.dope.asSchemaArray
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.DEFAULT_ITERATOR_VARIABLE
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.Iterator
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.IteratorManager
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.SatisfiesType
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.SatisfiesType.ANY
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.SatisfiesType.EVERY
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.any
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.every
+import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.DopeSchemaArray
+import ch.ergon.dope.validtype.NumberType
+import ch.ergon.dope.validtype.StringType
+import com.schwarz.crystalapi.schema.CMField
+import com.schwarz.crystalapi.schema.CMList
+import com.schwarz.crystalapi.schema.CMObject
+import com.schwarz.crystalapi.schema.CMObjectList
+import com.schwarz.crystalapi.schema.CMType
+import com.schwarz.crystalapi.schema.Schema
+
+sealed class SatisfiesSchemaExpression<S : Schema>(
+    private val satisfiesType: SatisfiesType,
+    private val variable: String,
+    private val list: DopeSchemaArray<S>,
+    private val predicate: (SchemaIterator<S>) -> TypeExpression<BooleanType>,
+) : TypeExpression<BooleanType> {
+    override fun toDopeQuery(): DopeQuery {
+        val iteratorVariable = if (variable == DEFAULT_ITERATOR_VARIABLE) {
+            variable + IteratorManager.count
```

Comment:
Just for my own understanding: Does this need to be the same `IteratorManager` as in the `SatisfiesExpression.kt` file or could it be its own separate one?

Reply:
It could be in its own file, but will be refactored in DOPE-218, where we will combine the two Managers (Iterator and Parameter).